hsds/attr_sn.py

```diff
@@ -764,3 +1008,217 @@ async def PUT_AttributeValue(request):
     resp = await jsonResponse(request, req_rsp)
     log.response(request, resp=resp)
     return resp
+
+
+async def _get_attributes(app, obj_id, attr_names,
+                          IncludeData=False,
+                          ignore_nan=False,
+                          bucket=None
+                          ):
+    """ get the requested set of attributes from the object"""
+    collection = getCollectionForId(obj_id)
+    node_url = getDataNodeUrl(app, obj_id)
+    req = f"{node_url}/{collection}/{obj_id}/attributes"
+    log.debug(f"POST Attributes: {req}")
+    params = {}
+    if IncludeData:
+        params["IncludeData"] = 1
+    if ignore_nan:
+        params["ignore_nan"] = 1
+    if bucket:
+        params["bucket"] = bucket
+    data = {"attributes": attr_names}
+    log.debug(f"using params: {params}")
+    dn_json = await http_post(app, req, data=data, params=params)
+    log.debug(f"got attributes json from dn for obj_id: {dn_json}")
+
+    if len(dn_json) < len(attr_names):
```

Comment:
Should this check be reworked to `if len(dn_json['attributes']) != len(attr_names)`, and placed after the check that `'attributes'`is in `dn_json`?

Reply:
fixed